my-react-app/src/components/Form/Input.tsx

```diff
@@ -0,0 +1,60 @@
+import clsx from "clsx";
+import { forwardRef, LegacyRef } from "react";
+import { FieldErrors, FieldValues, UseFormRegister } from "react-hook-form";
+import CountryList from "./CountryList";
+
+interface Props {
+  type: string;
+  placeholder: string;
+  name: string;
+  errors?: FieldErrors<FormData>;
+  register: UseFormRegister<FieldValues>;
+  required: boolean;
+  addClass?: string;
+  CountryListBool?: boolean;
+}
+const Input = forwardRef<HTMLInputElement, Props>(
```

Comment:
richtig geile componente die du da zum ersten mal gebaut hast !! 

 wenn du es noch zu 100% perfekt machen willst dann schau dir mal an was das register von reacht hook forms alles zurück liefert diese props und funktionen kannst dann hier in der schnittstelle mit anbieten.. dann brauchst ned immer das register von react hook form mit rein geben was man auch nicht sollte .. 
 
 warum sollte man das nicht ?? 
 
 falls wir zB mal entscheiden dass wir nicht react hook form nehmen sondern ein ähnliches packages kann es sein dass es dieses register in einer anderen form gibt. bsp. mantine heist es getInputProps
 
 also die einzelene props von register hier raus geben und draussen mit der alten schreibweise {...register('name')} anwenden